FloatType.h

```diff
@@ -16,8 +19,18 @@ struct FloatType
     FloatType& multiply( float rhs );
     FloatType& divide(  float rhs );
 
+    FloatType& pow(const IntType& i);
+    FloatType& pow(const FloatType& f);
+    FloatType& pow(const DoubleType& d);
+    FloatType& pow(float f);
+
+    //Operators for Static Conversion
+    //Const objects can only call their const member functions
     operator float()  {return *floatPtr;}
```

Comment:
You don't need a non-const version of this, since none of your functions take a non-const version.
- [ ] remove